pages/utils.py

```diff
@@ -0,0 +1,37 @@
+import datetime
+import logging
+from time import sleep
+
+
+def wait_until_ok(timeout=5, period=0.25):
+    """Retries function until ok"""
+    logger = logging.getLogger("[WaitUntilOk]")
+
+    def decorator(original_function):
+
+        def wrapper(*args, **kwargs):
+            end_time = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
+            while True:
+                try:
+                    return original_function(*args, **kwargs)
+                except Exception as err:
+                    if datetime.datetime.now() > end_time:
+                        logger.warning(f"Catch: {err}")
+                        raise err
+                    sleep(period)
+
+        return wrapper
+
+    return decorator
+
+
+def log_wrapper(func):
```

Comment:
Hint: You don't have to put it in your code if you don't use it

Reply:
Thanks. I tried to add an example.